src/app/database/database.component.ts

```diff
@@ -15,23 +15,35 @@ import { DataProviderFactoryService } from '../providers/data-provider-factory.s
 export class DatabaseComponent implements OnInit {
     words: WordEntry[];
     languages: string[];
+    languagePair: LanguagePair;
     primaryLanguage: string;
     secondaryLanguage: string;
 
     private languageIndexer: LanguageIndexer;
 
     constructor(
-            dataProviderFactory: DataProviderFactoryService,
+            private dataProviderFactory: DataProviderFactoryService,
             private wordsDatabaseService: WordsDatabaseService,
             private settingsService: SettingsService) {
         this.languageIndexer = dataProviderFactory.dataProviderInUse().retrieveLanguageIndexer();
     }
 
     ngOnInit() {
-        const languagePair = this.settingsService.languagePairInUse();
-        this.words = this.wordsDatabaseService.wordsFor(languagePair);
-        this.primaryLanguage = this.languageIndexer.nameOf(languagePair.src);
-        this.secondaryLanguage = this.languageIndexer.nameOf(languagePair.dst);
+        this.languagePair = this.settingsService.languagePairInUse();
+        this.words = this.wordsDatabaseService.wordsFor(this.languagePair);
+        this.primaryLanguage = this.languageIndexer.nameOf(this.languagePair.src);
+        this.secondaryLanguage = this.languageIndexer.nameOf(this.languagePair.dst);
         this.languages = this.languageIndexer.allNames();
     }
+
+    addWordEntry(wordEntry: WordEntry) {
+        // TODO: probably need to do add the word in one place instead of two:
```

Comment:
Please pay attention to these 2 lines: how can i improve the design? (in the future the wordsDatabase will probably subscribe to the changes in DataProvider)